File: task-03/Go/subtask-03.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func printStarPattern(n int) {
	for i := 1; i <= n; i += 2 {
		padding := strings.Repeat(" ", (n-i)/2)
		stars := strings.Repeat("*", i)
		fmt.Println(padding + stars)
	}

	for i := n - 2; i >= 1; i -= 2 {
		padding := strings.Repeat(" ", (n-i)/2)
		stars := strings.Repeat("*", i)
		fmt.Println(padding + stars)
	}
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter a number: ")
	input, _ := reader.ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(input))

	printStarPattern(n)
}

File: task-03/Python/subtask-03.py
def print_star_pattern(n):

    for i in range(1, n + 1, 2):
        print(' ' * ((n - i) // 2), end='')
        print('*' * i)


    for i in range(n-2, 0, -2):
        print(' ' * ((n - i) // 2), end='')
        print('*' * i)

n = int(input("Enter a number"))
print_star_pattern(n)

